src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,960 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+import glob
+import json
+import os
+import random
+import sys
+import string
+import tempfile
+import time
+import unittest
+from core.src.bootstrap.Constants import Constants
+from core.tests.library.ArgumentComposer import ArgumentComposer
+from core.tests.library.RuntimeCompositor import RuntimeCompositor
+
+
+class TestStatusHandlerTruncation(unittest.TestCase):
+    def setUp(self):
+        self.runtime = RuntimeCompositor(ArgumentComposer().get_composed_arguments(), True)
+        self.container = self.runtime.container
+
+    def tearDown(self):
+        self.runtime.stop()
+
+    def test_assessment_patches_under_size_limit_not_truncated(self):
+        """ Perform no truncation on assessment patches.
+        Input: xxx assessment patches
+        Output:
+        operation: Assessment,
+        assessment substatus name: PatchAssessmentSummary,
+        completed status file < 126kb,
+        size of status file before truncation == size of status file after truncation,
+        assessment substatus status: success,
+        assessment substatus patches == patch_count,
+        no assessment tombstone records,
+        assessment errors code: 0 (success),
+        assessment errors count: 0,
+        assessment errors details code: 0 (success). """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = 500
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+
+        assessment_msg = json.loads(complete_substatus_file_data[0]["status"]["substatus"][0]["formattedMessage"]["message"])
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count, assessment_msg, is_truncated=False)
+
+        # Assert no truncated status file
+        self.__assert_no_truncation_status_file(complete_substatus_file_data, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count)
```

Comment:
A lot of the asserts between self.__assert_patch_summary_from_status() and self.__assert_no_truncation_status_file() are the same. What is the reason for this double validation?

Please review all such similar cases

Reply:
combine assertions into __assert_patch_summary_from_status.

done